### ISSUE TITLE
Refs only get one item

### ISSUE THREAD

**Issue by elmarburke** [infinum/mobx-jsonapi-store#29]
Hi, I'm using `"mobx-jsonapi-store": "^3.5.0",`. Currently I have the problem, that my refs only get one object while there are two include in the request and also in the store.

I call `store.fetchAll('organizations', undefined, {include: 'units'});` to get all the data.

and this is my store. In this, i had to quickfix fetch since some context got lost.

I guess, i've overlooked something.

```typescript
import { Record, Store, config } from 'mobx-jsonapi-store';

config.fetchReference = (req: RequestInfo) => window.fetch(req); // Fix for window error
config.baseUrl = 'http://localhost:5000/';

class UnitRecord extends Record {
  static type = 'units';
  static refs = { organization: 'organizations' };

  organization?: OrganizationRecord;
}

export class OrganizationRecord extends Record {
  static type = 'organizations';
  static refs = { units: 'units' };
  
  units?: OrganizationRecord|OrganizationRecord[];

  name: string;
}

export class ApiStore extends Store {
  static types = [OrganizationRecord, UnitRecord];

  organizations: Array<OrganizationRecord>;
  units: Array<UnitRecord>;
}

const store = window['__store'] = new ApiStore();

export default store;
```

**Comment by DarkoKukovec** [comment]
Hi @elmarburke. On the first glance everything seems fine in the code. Can you add the response example that doesn't work correctly?

**Comment by elmarburke** [comment]
Sure, this is the response from `GET http://localhost:5000/organizations?include=units`

```json
{  
  "jsonapi":{  
    "version":"1.0"
  },
  "meta":{  
    "page":{  
      "offset":0,
      "limit":50,
      "total":1
    }
  },
  "links":{  
    "self":"http://localhost:5000/organizations?include=units"
  },
  "data":[  
    {  
      "type":"organizations",
      "id":"ORG-A",
      "attributes":{  
        "name":"Organization A"
      },
      "links":{  
        "self":"http://localhost:5000/organizations/ORG-A"
      },
      "relationships":{  
        "units":{  
          "meta":{  
            "relation":"foreign",
            "belongsTo":"units",
            "as":"organization",
            "many":true,
            "readOnly":true
          },
          "links":{  
            "self":"http://localhost:5000/units/relationships/?organization=ORG-A",
            "related":"http://localhost:5000/units/?filter[organization]=ORG-A"
          }
        }
      }
    }
  ],
  "included":[  

  ]
}
```

**Comment by elmarburke** [comment]
Sure, this is the `GET http://localhost:5000/organizations?include=units`

```json
{  
  "jsonapi":{  
    "version":"1.0"
  },
  "meta":{  
    "page":{  
      "offset":0,
      "limit":50,
      "total":1
    }
  },
  "links":{  
    "self":"http://localhost:5000/organizations?include=units"
  },
  "data":[  
    {  
      "type":"organizations",
      "id":"ORG-A",
      "attributes":{  
        "name":"Organization A"
      },
      "links":{  
        "self":"http://localhost:5000/organizations/ORG-A"
      },
      "relationships":{  
        "units":{  
          "meta":{  
            "relation":"foreign",
            "belongsTo":"units",
            "as":"organization",
            "many":true,
            "readOnly":true
          },
          "links":{  
            "self":"http://localhost:5000/units/relationships/?organization=ORG-A",
            "related":"http://localhost:5000/units/?filter[organization]=ORG-A"
          }
        }
      }
    }
  ],
  "included":[  
    {  
      "type":"units",
      "id":"ORG-A-01",
      "attributes":{  
        "name":"Unit 01"
      },
      "links":{  
        "self":"http://localhost:5000/units/ORG-A-01"
      },
      "relationships":{  
        "organization":{  
          "meta":{  
            "relation":"primary",
            "readOnly":false
          },
          "links":{  
            "self":"http://localhost:5000/units/ORG-A-01/relationships/organization",
            "related":"http://localhost:5000/units/ORG-A-01/organization"
          },
          "data":{  
            "type":"organizations",
            "id":"ORG-A"
          }
        },
        "orders":{  
          "meta":{  
            "relation":"foreign",
            "belongsTo":"orders",
            "as":"unit",
            "many":true,
            "readOnly":true
          },
          "links":{  
            "self":"http://localhost:5000/orders/relationships/?unit=ORG-A-01",
            "related":"http://localhost:5000/orders/?filter[unit]=ORG-A-01"
          }
        }
      }
    },
    {  
      "type":"units",
      "id":"ORG-A-02",
      "attributes":{  
        "name":"Unit 02"
      },
      "links":{  
        "self":"http://localhost:5000/units/ORG-A-02"
      },
      "relationships":{  
        "organization":{  
          "meta":{  
            "relation":"primary",
            "readOnly":false
          },
          "links":{  
            "self":"http://localhost:5000/units/ORG-A-02/relationships/organization",
            "related":"http://localhost:5000/units/ORG-A-02/organization"
          },
          "data":{  
            "type":"organizations",
            "id":"ORG-A"
          }
        },
        "orders":{  
          "meta":{  
            "relation":"foreign",
            "belongsTo":"orders",
            "as":"unit",
            "many":true,
            "readOnly":true
          },
          "links":{  
            "self":"http://localhost:5000/orders/relationships/?unit=ORG-A-02",
            "related":"http://localhost:5000/orders/?filter[unit]=ORG-A-02"
          }
        }
      }
    }
  ]
}
```

**Comment by DarkoKukovec** [comment]
Hm... I didn't think about this use case where the main model doesn't have a reference to included models but the other way around.

JSON API spec doesn't seem to cover this use case, so I'll need to think about what's the best way to implement this.

A (not ideal) workaround: Remove the units ref from OrganizationRecord, and add a computed property instead:

```typescript
@computed get units(): UnitRecord[] {
  return (this.__collection.findAll('units') as UnitRecord[])
    .filter((unit: UnitRecord) => unit.organization === this);
}
```

**Comment by DarkoKukovec** [comment]
Should be fixed with version 3.6.0: 
```typescript
class OrganizationRecord extends Record {
    public static type = 'organizations';
    public static refs = {
      units: {
        model: 'units',
        property: 'organization',
      },
    };

    public name: string;
    public units: Array<UnitRecord>;
  }
```

Let me know if there are any other issues.
